bundles/org.openhab.binding.noolite/src/main/java/org/openhab/binding/noolite/handler/NooliteMTRF64BridgeHandler.java

```diff
@@ -0,0 +1,206 @@
+        if (thingHandlerMap.get(thingID) == null) {
+            thingHandlerMap.put(thingID, thingHandler);
+            logger.debug("register thingHandler for thing: {}", thingHandler);
+            updateThingHandlerStatus(thingHandler, this.getStatus());
+            if (thingID.equals("localhost.")) {
+                updateThingHandlerStatus(thingHandler, ThingStatus.OFFLINE);
+            }
+        } else {
+            logger.debug("thingHandler for thing: '{}' already registerd", thingID);
+            updateThingHandlerStatus(thingHandler, this.getStatus());
+        }
+    }
+
+    public ThingStatus getStatus() {
+        return getThing().getStatus();
+    }
+
+    private void updateThingHandlerStatus(NooliteHandler thingHandler, ThingStatus status) {
+        thingHandler.updateStatus(status);
+    }
+
+    @SuppressWarnings("null")
+    public static void updateValues(byte[] data) {
+        String thingID = data[1] + "." + data[4];
+        nooliteHandler = thingHandlerMap.get(thingID);
+        nooliteHandler.updateValues(data);
+    }
+
+    @SuppressWarnings("null")
+    public synchronized void sendMessage(NooliteHandler nooliteHandler, @Nullable ChannelUID channel, Command command) {
+        logger.debug("{}", command);
+
+        byte[] data = new byte[17];
+        data[0] = (byte) 0b10101011;
+        data[1] = (byte) Integer.parseInt(nooliteHandler.getThing().getConfiguration().get("type").toString());
+        data[2] = 0;
+        data[3] = 0;
+        if ((channel != null) && (channel.getId().equals(NooLiteBindingConstants.CHANNEL_BINDCHANNEL))
+                && !(command.toString().equals("REFRESH"))) {
+            if (Integer.parseInt(nooliteHandler.getThing().getConfiguration().get("type").toString()) == 3) {
+                data[2] = 3;
+            }
+            data[4] = (byte) (Integer.parseInt(command.toString()));
+            data[5] = 15;
+            data[6] = 0;
+            data[7] = 0;
+            data[8] = 0;
+            data[9] = 0;
+            data[10] = 0;
+        } else {
+            data[4] = (byte) (Integer.parseInt(nooliteHandler.getThing().getConfiguration().get("port").toString()));
+        }
+        if (command.toString().equals("ON")) {
+            data[5] = 2;
+            data[6] = 0;
+            data[7] = 0;
+            data[8] = 0;
+            data[9] = 0;
+            data[10] = 0;
+        } else if (command.toString().equals("OFF")) {
+            data[5] = 0;
+            data[6] = 0;
+            data[7] = 0;
+            data[8] = 0;
+            data[9] = 0;
+            data[10] = 0;
+        }
+
+        data[11] = 0;
+        data[12] = 0;
+        data[13] = 0;
+        data[14] = 0;
+
+        short sum = 0;
+        for (int i = 0; i < 14; i++) {
+            sum += data[i];
+        }
+
+        data[15] = (byte) sum;
+        data[16] = (byte) 0b10101100;
+        if (!command.toString().equals("REFRESH")) {
```

Comment:
If you don't do anything when the command is refresh you should do this as a first test. Otherwise all the code above is run for nothing.
```suggestion
        if (!(command instanceof RefreshType)) {
```